### PR TITLE
ref(ts): Convert ActionLink

### PR DESCRIPTION
**Pull request by leedongwei**
Typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/actions/actionLink.tsx [hunk 1]
```diff
@@ -47,16 +45,22 @@ export default class ActionLink extends React.Component {
...
       );
     }
   }
 }
+
+const ActionLinkAnchor = styled('a')<{
+  disabled?: boolean;
+}>`
+  pointer-events: none;
+`;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Doesn't look like you're using the `disabled` prop in here?

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Ah good catch.

File: src/sentry/static/sentry/app/components/actions/actionLink.tsx [hunk 2]
```diff
@@ -47,16 +45,22 @@ export default class ActionLink extends React.Component {
...
         >
           {children}
-        </a>
+        </ActionLinkAnchor>
       );
     }
   }
 }
+
+const ActionLinkAnchor = styled('a')<{
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Kind of a style nit, but I've just been writing these as single lines

```jsx
const ActionLinkAnchor = styled('a')<{disabled?: boolean;}>`
...
`
```

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/actions/actionLink.tsx]: 👍

File: src/sentry/static/sentry/app/components/actions/actionLink.tsx (hunk 1 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Fixed but Github is locking on the wrong line of code.

File: src/sentry/static/sentry/app/components/actions/actionLink.tsx [hunk 3]
```diff
@@ -1,21 +1,19 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-
+import styled from 'react-emotion';
 import classNames from 'classnames';
 
 import Confirm from 'app/components/confirm';
 
-export default class ActionLink extends React.Component {
-  static propTypes = {
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Should we retain the propTypes for JSX usage of this component?

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/actions/actionLink.tsx]: Sure. I've added it back in.

**Review by evanpurkhiser — APPROVED** [review]

**Comment by leedongwei** [comment]
Latest commit ae43177 updates failing snapshots - no new code was added.

**Pull request merged**